### PR TITLE
[feature] allow uncaching of other media types

### PR DESCRIPTION
**Pull request by NyaaaWhatsUpDoc**

### CODE REVIEW

**Comment by NyaaaWhatsUpDoc** [comment]
The failing test I'm pretty sure is due to me fiddling with the test models for remote media tests, something i'll take a look at another day

File: internal/media/prunemeta.go [hunk 1]
```diff
@@ -28,17 +29,23 @@ import (
...
+			break
+		}
 
-	// select 20 attachments at a time and prune them
-	for attachments, err = m.db.GetAvatarsAndHeaders(ctx, maxID, selectPruneLimit); err == nil && len(attachments) != 0; attachments, err = m.db.GetAvatarsAndHeaders(ctx, maxID, selectPruneLimit) {
 		// use the id of the last attachment in the slice as the next 'maxID' value
-		l := len(attachments)
-		log.Tracef("PruneAllMeta: got %d attachments with maxID < %s", l, maxID)
-		maxID = attachments[l-1].ID
+		log.Tracef("PruneAllMeta: got %d attachments with maxID < %s", len(attachments), maxID)
```
Comment (tsmethurst) [inline internal/media/prunemeta.go:47]: can remove PruneAllMeta from this line now right?

Comment (tsmethurst) [reply internal/media/prunemeta.go:47]: Same comment applies for other trace + debug logs in this PR too. Not critical though.

**Review by tsmethurst — COMMENTED** [review]
Huh, nice! I'm pleasantly surprised by how few code changes this ended up requiring. So the refetch of avatars + headers Just Works now?

File: internal/media/prunemeta.go (hunk 1 above)
Comment (NyaaaWhatsUpDoc) [reply internal/media/prunemeta.go:47]: I actually have another PR lined up that cleans this up if that's all good :)

**Comment by NyaaaWhatsUpDoc** [comment]
> Huh, nice! I'm pleasantly surprised by how few code changes this ended up requiring. So the refetch of avatars + headers Just Works now?

it does! confirmed by the fact that before https://github.com/superseriousbusiness/gotosocial/pull/1234/commits/899d3859158dcf92e271e55d42f1547a43cc9717 i accidentally nuked all of my remote avatars/headers and they all reloaded :')

**Comment by NyaaaWhatsUpDoc** [comment]
@tsmethurst all tests are passing and this is ready to go now :)

turns out i inserted the generated preserialized key in the test account models area at the wrong index.

**Comment by tsmethurst** [comment]
Sweet! Gonna try it out :D

**Pull request merged**